// montage-wordpress/core/wordpress-connector.js
var Montage = require("montage/core/core").Montage,
    Reqwest = require("reqwest");

exports.WordpressConnector = Montage.specialize({

    constructor: {
        value: function WordpressConnector () {
        }
    },

    host: {
        value: null
    },

    path: {
        value: null
    },
    
    categoryFilter: {
        value: null
    },
    
    response: {
        value: null
    },

    makeRequest: {
        value: function() {
            var url = "http://" + this._host + "/wp-json/" + this._path + "?_jsonp=cb",
                self = this,
                latestRequest,
                query;
            
            if (this.categoryFilter) {
                query = {"filter[category_name]": this.categoryFilter};
                url += "&" + Reqwest.toQueryString(query);
            }

            this._latestRequest = Reqwest({url: url, type: "jsonp", method: "get",
                jsonpCallback: "_jsonp", jsonpCallbackName: "cb"});
            
            response
                .then(function (value) {
                    
                });
        }
    },

})
